Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -228,6 +228,33 @@ void AssignmentExpression::generate_bytecode(Bytecode::Generator& generator) con
         }
 
         m_lhs->generate_bytecode(generator);
+
+        auto& rhs_block = generator.make_block();
+        auto& end_block = generator.make_block();
+
+        // Logical assignments short circuit.
+        if (m_op == AssignmentOp::AndAssignment) { // &&=
+            generator.emit<Bytecode::Op::JumpConditional>().set_targets(
+                Bytecode::Label { rhs_block },
+                Bytecode::Label { end_block });
+        } else if (m_op == AssignmentOp::OrAssignment) { // ||=
+            generator.emit<Bytecode::Op::JumpConditional>().set_targets(
+                Bytecode::Label { end_block },
+                Bytecode::Label { rhs_block });
+        } else if (m_op == AssignmentOp::NullishAssignment) { // ??=
+            generator.emit<Bytecode::Op::JumpNullish>().set_targets(
+                Bytecode::Label { rhs_block },
+                Bytecode::Label { end_block });
+        } else {
+            generator.emit<Bytecode::Op::Jump>().set_targets(
+                Bytecode::Label { rhs_block },
+                {});
```

Comment:
Just realised I forgot this and just added it in.
It's a bit unfortunate the way this is done, is there a way to avoid this?

Reply:
Sure there is, just don't generate the basic blocks if it's none of these
e.g. look at how `ForStatement` collapses blocks that are going to be empty